### PR TITLE
[CPDLP-3199] TS8 - Add outcomes logic to void declaration endpoint

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

Ticket: https://dfedigital.atlassian.net.mcas.ms/browse/CPDLP-3199

### Changes proposed in this pull request

* Created `ParticipantOutcomes::Void` service
* Updated `Declarations::Void` to run `ParticipantOutcomes::Void`

### CODE REVIEW

File: app/models/course_group.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ class CourseGroup < ApplicationRecord
 
   validates :name, presence: { message: "Enter a unique course group name" }, uniqueness: { message: "Course name already exist, enter a unique name" }
 
+  scope :leadership_or_specialist, -> { where(name: %w[leadership specialist]) }
```
Comment (leandroalemao) [inline app/models/course_group.rb:7]: Hi @mooktakim Could you please add a spec for this scope? I followed your suggestion on the other PR and added a spec in there.. so whatever PR gets merged first, the scope will be in.. tks
https://github.com/DFE-Digital/npq-registration/pull/1499/files#diff-be233ac6eae795dbfb4a1a24629eef68a767d6875377e03a5c7d1b7cc84c3dd6R80

Comment (mooktakim) [reply app/models/course_group.rb:7]: @leandroalemao i was planning on merging your ticket first as its almost ready to be merged. 👍

Comment (mooktakim) [reply app/models/course_group.rb:7]: I'll copy in the test anyway, it should be easy to merge.

Comment (cwrw) [reply app/models/course_group.rb:7]: This has been added now in three prs, @leandroalemao maybe we can open a separate pr for this scope and merge so the others can rebase those changes?

**Review by cwrw — APPROVED** [review]

File: app/models/course_group.rb (hunk 1 above)
Comment (leandroalemao) [reply app/models/course_group.rb:7]: 👍 I've got a PR to merge tomorrow which includes this changes .. so it should be good after that.

**Pull request merged**